### PR TITLE
optional max_guesses (intended usage '!hangman ruby 6')

### PR DESCRIPTION
**Pull request by rthbound**
I'd like to see this feature added so we can scale the difficulty as desired...

I'll be glad to amend changes and squash as needed.

### CODE REVIEW

**Comment by jhass** [comment]
Looks good, the regex will make it reject negative values, but how about an upper limit?

File: bot/src/plugins/hangman.cr [hunk 1]
```diff
@@ -134,10 +139,10 @@ class Hangman
     end
   end
 
-  def start_game msg, list
+  def start_game msg, list, guess_max
     unless GAMES.has_key? msg.channel.name
       if Game::WORDLISTS.has_key? list
-        GAMES[msg.channel.name] = Game.new list
+        GAMES[msg.channel.name] = Game.new list, [guess_max, Game::DEFAULT_GUESS_MAX].min
```
Comment (jhass) [inline bot/src/plugins/hangman.cr]: Don't need to allocate an array here, just use a tuple: `{guess_max, Game::DEFAULT_GUESS_MAX}.min`

**Comment by rthbound** [comment]
Thanks @jhass ... modified to enforce an upper limit by taking the minimum value from the tuple `{guess_max, Game::DEFAULT_GUESS_MAX}`

File: bot/src/plugins/hangman.cr [hunk 2]
```diff
@@ -114,9 +116,12 @@ class Hangman
       list msg
     when /^!hangman\s+\w+$/
       _c, list = command.split
-      start_game msg, list
+      start_game msg, list, Game::DEFAULT_GUESS_MAX
+    when /^!hangman\s+\w+\s+\d+$/
+      _c, list, guess_max = command.split
+      start_game msg, list, guess_max
```
Comment (jhass) [inline bot/src/plugins/hangman.cr]: `guess_max.to_i`

File: bot/src/plugins/hangman.cr [hunk 3]
```diff
@@ -134,10 +139,10 @@ class Hangman
     end
   end
 
-  def start_game msg, list
+  def start_game msg, list, guess_max
```
Comment (rthbound) [inline bot/src/plugins/hangman.cr]: @jhass  Also `def start_game msg, list, guess_max : Number`?

Comment (jhass) [inline bot/src/plugins/hangman.cr]: Nah, I prefer to not restrict unless needed for some reason or when providing external API.

Comment (rthbound) [inline bot/src/plugins/hangman.cr]: 10-4

**Comment by jhass** [comment]
Thanks!

**Pull request merged**